### ISSUE TITLE
[ @typescript-eslint/no-unused-expressions] false positive on optional chaining

### ISSUE THREAD

**Issue by clement9527** [typescript-eslint/typescript-eslint#2475]
- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    " @typescript-eslint/no-unused-expressions": ["error"]
  }
}
```

```TS
import React, { FC, useRef } from 'react';

const SomeComponent: FC = (props) => {
  const ref = useRef<HTMLDivElement>(null);
  const test = () => {
    ref.current?.click();
  };
  return <div ref={ref}>TODO</div>;
};
export default SomeComponent;

```

**Expected Result**
No lint error

**Actual Result**
./src/component/SomeComponent.tsx
  Line 6:5:  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `Eslint` | `6.8.0` |

**Comment by bradzacher** [comment]
what version of ESLint are you using? It's part of the template on purpose because it's important information.

As per the top comment in the issue template:

```
<!--
Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->
```

**Comment by clement9527** [comment]
The project is not using ESLint directly but use prettier. 

`  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json',
  },
  plugins: ['prettier', '@typescript-eslint', 'react', 'flowtype'],
  extends: [
    'airbnb-typescript',

    /* keep prettier at the bottom */
    'prettier',
    'prettier/react',
    'prettier/flowtype',
    'prettier/@typescript-eslint',
  ],
`

**Comment by bradzacher** [comment]
> The project is not using ESLint directly

you must be using eslint, or else you wouldn't be able to lint your codebase.
what version is the `eslint` package you have installed?

**Comment by clement9527** [comment]
Sorry, it was installed globally, it is 6.14.7.  I have updated the original thread with the Eslint version

**Comment by bradzacher** [comment]
> 6.14.7

ESLint hasn't ever released this version. Are you sure that's correct?

**Comment by clement9527** [comment]
7.8.1

**Comment by clement9527** [comment]
Sorry, I have copied my npm version incorrectly. I think it is `6.8.0` as it is transient dep of `prettier-eslint`. I don't have eslint in my package.json. 

```"prettier-eslint": {
      "version": "11.0.0",
      "resolved": "https://registry.npmjs.org/prettier-eslint/-/prettier-eslint-11.0.0.tgz",
      "integrity": "sha512-ACjL7T8m10HCO7DwYdXwhNWuZzQv86JkZAhVpzFV9brTMWi3i6LhqoELFaXf6RetDngujz89tnbDmGyvDl+rzA==",
      "dev": true,
      "requires": {
        "@typescript-eslint/parser": "^3.0.0",
        "common-tags": "^1.4.0",
        "dlv": "^1.1.0",
        "eslint": "^6.8.0",
        "indent-string": "^4.0.0",
        "lodash.merge": "^4.6.0",
        "loglevel-colored-level-prefix": "^1.0.0",
        "prettier": "^2.0.0",
        "pretty-format": "^23.0.1",
        "require-relative": "^0.8.7",
        "typescript": "^3.9.3",
        "vue-eslint-parser": "~7.1.0"
      },

**Comment by bradzacher** [comment]
`7.8.1` will support optional chaining just fine.
But if you're using `prettier-eslint` to do your lint runs, it'll use v6 of eslint, because it hasn't been upgraded: https://github.com/prettier/prettier-eslint/blob/06d40ecdd43feefb8fc8479f93b9086291e6697d/package.json#L24

**Comment by clement9527** [comment]
That would explain it! Thanks Brad!

**Comment by esetnik** [comment]
@bradzacher should this be reopened as it's a compatibility issue with eslint v6? I faced the same issue and i'm also stuck on eslint v6 but not using prettier-eslint.

**Comment by bradzacher** [comment]
Nope - this is an intentional breaking change.
We won't be duplicating code from the base ESLint codebase in order to support this retroactively.
This is specifically called out in the https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0 notes:

> This means a few things right now:
> - The ESLint core rules will have complete support for optional chaining, without need for extension rules 🎉
>    - ***you will need to upgrade to at least ESLint 7.5.0 in order to have support in lint rules***
